extra/modules/greenbids-real-time-data/src/main/java/org/prebid/server/hooks/modules/greenbids/real/time/data/v1/GreenbidsRealTimeDataProcessedAuctionRequestHook.java

```diff
@@ -0,0 +1,219 @@
+    private final Cache<String, ThrottlingThresholds> thresholdsCacheWithExpiration;
+    private final String gcsBucketName;
+    private final String onnxModelCacheKeyPrefix;
+    private final String thresholdsCacheKeyPrefix;
+    private final Storage storage;
+    private final File database;
+
+    public GreenbidsRealTimeDataProcessedAuctionRequestHook(
+            ObjectMapper mapper,
+            Cache<String, OnnxModelRunner> modelCacheWithExpiration,
+            Cache<String, ThrottlingThresholds> thresholdsCacheWithExpiration,
+            String gcsBucketName,
+            String onnxModelCacheKeyPrefix,
+            String thresholdsCacheKeyPrefix,
+            Storage storage,
+            File database) {
+        this.mapper = Objects.requireNonNull(mapper);
+        this.jacksonMapper = new JacksonMapper(mapper);
+        this.modelCacheWithExpiration = modelCacheWithExpiration;
+        this.thresholdsCacheWithExpiration = thresholdsCacheWithExpiration;
+        this.gcsBucketName = gcsBucketName;
+        this.onnxModelCacheKeyPrefix = onnxModelCacheKeyPrefix;
+        this.thresholdsCacheKeyPrefix = thresholdsCacheKeyPrefix;
+        this.storage = storage;
+        this.database = database;
+    }
+
+    @Override
+    public Future<InvocationResult<AuctionRequestPayload>> call(
+            AuctionRequestPayload auctionRequestPayload,
+            AuctionInvocationContext invocationContext) {
+        final AuctionContext auctionContext = invocationContext.auctionContext();
+        final BidRequest bidRequest = auctionContext.getBidRequest();
+        final Partner partner = parseBidRequestExt(bidRequest);
+
+        if (partner == null) {
+            return Future.succeededFuture(toInvocationResult(
+                    bidRequest, null, InvocationAction.no_action));
+        }
+
+        OnnxModelRunner onnxModelRunner = null;
+        ThrottlingThresholds throttlingThresholds = null;
+        Map<String, Map<String, Boolean>> impsBiddersFilterMap = null;
+        try {
+            final OnnxModelRunnerWithThresholds onnxModelRunnerWithThresholds = new OnnxModelRunnerWithThresholds(
+                    jacksonMapper,
+                    modelCacheWithExpiration,
+                    thresholdsCacheWithExpiration,
+                    storage,
+                    gcsBucketName,
+                    onnxModelCacheKeyPrefix,
+                    thresholdsCacheKeyPrefix);
+            onnxModelRunner = onnxModelRunnerWithThresholds.retrieveOnnxModelRunner(partner);
+            throttlingThresholds = onnxModelRunnerWithThresholds.retrieveThreshold(partner);
+
+            if (onnxModelRunner == null || throttlingThresholds == null) {
+                throw new PreBidException("Cache was empty, fetching and put artefacts for next request");
+            }
+
+            final GreenbidsInferenceData greenbidsInferenceData = new GreenbidsInferenceData(jacksonMapper, database);
```

Comment:
I believe this one is also can be created once

P.S. I've already asked you to check which objects can be created only once, so once more time, double-check the code and define such objects as beans

Reply:
In this one I've modified the class with @Builder annotation and static`prepareData` that returns the built GreenbidsInferenceData, because we need to recreate it at each call with different `throttlingMessages`, `throttlingInferenceRows` and these fields are private and final. So `GreenbidsInferenceData` has to be defined at each call

I've checked other object created ad hook call `GreenbidsInvocationResult`. Same as for previous class here we need to rebuilt the object with different private final `updatedBidRequest`, `invocationAction`, `analyticsResult`